imod/mf6/drn.py

```diff
@@ -155,3 +169,136 @@ def _validate(self, schemata, **kwargs):
+            as present imod 5 and the second one the input array
+        allocation_methods: dict[str, ALLOCATION_OPTION]], optional
+            allocation option per package. The key is the package name as present in
+            imod 5. All or none should be present.
+        distributing_option: dict[str, DISTRIBUTING_OPTION]], optional
+            distributing option option per package. The key is the package name as present in
+            imod 5. All or none should be present.
+
+        Returns
+        -------
+        A list of Modflow 6 Drainage packages.
+        """
+
+        drainage_keys = [k for k in imod5_data.keys() if k[0:3] == "drn"]
+        if allocation_methods is None:
+            allocation_methods = dict.fromkeys(
+                drainage_keys, ALLOCATION_OPTION.first_active_to_elevation
+            )
+        if distributing_option is None:
+            distributing_option = dict.fromkeys(
+                drainage_keys, DISTRIBUTING_OPTION.by_corrected_transmissivity
+            )
+        if regridder_types is None:
+            regridder_types = dict.fromkeys(drainage_keys, Drainage._regrid_method)
+
+        drainage_packages = []
+        for key in drainage_keys:
+            package = cls._from_imod5_data_single(
+                key,
+                imod5_data,
+                target_discretization,
+                target_npf,
+                allocation_method=allocation_methods[key],
+                distributing_option=distributing_option[key],
+                regridder_types=regridder_types[key],
+            )
+            drainage_packages.append(package)
+        return drainage_packages
+
+    @classmethod
+    def _from_imod5_data_single(
+        cls,
+        key: str,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        target_discretization: StructuredDiscretization,
+        target_npf: NodePropertyFlow,
+        allocation_method: ALLOCATION_OPTION,
+        distributing_option: DISTRIBUTING_OPTION,
+        regridder_types: dict[str, tuple[RegridderType, str]],
+    ) -> "Drainage":
+        target_top = target_discretization.dataset["top"]
+        target_bottom = target_discretization.dataset["bottom"]
+        target_idomain = target_discretization.dataset["idomain"]
+
+        data = {
+            "elevation": imod5_data[key]["elevation"],
+            "conductance": imod5_data[key]["conductance"],
+        }
+
+        regridder_settings = deepcopy(cls._regrid_method)
+        if regridder_types is not None:
+            regridder_settings.update(regridder_types)
+
+        regrid_context = RegridderWeightsCache()
+
+        new_package_data = _regrid_package_data(
+            data, target_idomain, regridder_settings, regrid_context, {}
+        )
+
+        planar_elevation = new_package_data["elevation"]
+        if "layer" in new_package_data["elevation"].coords:
+            planar_elevation = new_package_data["elevation"].sel(layer=1, drop=True)
```

Comment:
I don't think the layer coordinate should be dropped in all cases. Only when the ``typing.grid.is_planar_grid`` helper function you created is true. So mainly when layer = 0

In the case where the drain package is assigned to layer 1 in the projectfile, this should be kept.